embedded/servo_driver/servo_driver.cc

```diff
@@ -32,46 +22,48 @@ ServoDriver::ServoDriver(const Config& config) {
   pwm_driver_->enable_auto_increment(true);
 
   try {
-    max_angle_ = config["max_angle"].as<float>();
-    calibrated_center_ = config["calibrated_center"].as<float>();
-    calibrated_max_ = config["calibrated_max"].as<float>();
+    max_angle_radians_ = config["max_angle_radians"].as<double>();
+    calibrated_min_pwm_count_ = config["calibrated_min_pwm_count"].as<uint>();
+    calibrated_max_pwm_count_ = config["calibrated_max_pwm_count"].as<uint>();
     servo_index_ = config["index"].as<int>();
     assert(servo_index_ >= 0);
   } catch (YAML::BadFile e) {
     std::string err = std::string("Could not find YAML file ");
     throw std::runtime_error(err);
   }
 
-  percentage_ = calibrated_center_;
-  set_percentage(percentage_);
+  set_angle_radians(0);
 }
 
-void ServoDriver::set_percentage(float unchecked_percentage) {
-  if (unchecked_percentage > 100 || unchecked_percentage < 0) {
+void ServoDriver::set_percentage(double unchecked_percentage, uint max_pwm_count, uint min_pwm_count) {
+  if (unchecked_percentage > 1 || unchecked_percentage < 0) {
     std::cout << "Desired servo percentage out of range: " << unchecked_percentage
               << std::endl;
   }
-  percentage_ = std::max(std::min(100.0f, unchecked_percentage), 0.0f);
+  percentage_ = std::max(std::min(1.0, unchecked_percentage), 0.0);
 
-  const float counts_range = MAX_COUNTS - MIN_COUNTS;
-  int counts = round(counts_range * percentage_ / 100 + MIN_COUNTS);
+  const uint counts_range = max_pwm_count - min_pwm_count;
+  const uint counts = counts_range * percentage_ + min_pwm_count;
+  pwm_count_ = counts;
   pwm_driver_->set_pwm(servo_index_, 0, counts);
 }
 
-float ServoDriver::get_percentage() const {
+double ServoDriver::get_percentage() const {
   return percentage_;
 }
 
+uint ServoDriver::get_pwm_count() const {
+  return pwm_count_;
+}
+
 int ServoDriver::get_servo_index() const {
   return servo_index_;
 }
 
-void ServoDriver::set_angle_radians(
-    float angle_radians) {  // TODO noble to give units to angle
-  float angle_degrees = rad_to_deg(angle_radians);
-  float angleFraction = static_cast<float>(angle_degrees) / max_angle_;
-  float half_range = (calibrated_max_ - calibrated_center_);
-  set_percentage(calibrated_center_ + half_range * angleFraction);
+void ServoDriver::set_angle_radians(double angle_radians) {
+  const double slope = 1.0 / (2*max_angle_radians_);
+  const double percentage = slope * (angle_radians + max_angle_radians_);
+  set_percentage(percentage, calibrated_max_pwm_count_, calibrated_min_pwm_count_);
```

Comment:
Add comments clarifying how to use this. We expect angles to be able to be negative?